server/commands/teamCreator.ts

```diff
@@ -54,27 +53,24 @@ export default async function teamCreator({
   // to the multi-tenant version, we want to restrict to a single team that MAY
   // have multiple authentication providers
   if (process.env.DEPLOYMENT !== "hosted") {
-    const teamCount = await Team.count();
+    const team = await Team.findOne();
 
-    // If the self-hosted installation has a single team and the domain for the
-    // new team is allowed then assign the authentication provider to the
-    // existing team
-    if (teamCount === 1 && domain && isDomainAllowed(domain)) {
-      const team = await Team.findOne();
-      invariant(team, "Team should exist");
-
-      authP = await team.$create<AuthenticationProvider>(
-        "authenticationProvider",
-        authenticationProvider
-      );
-      return {
-        authenticationProvider: authP,
-        team,
-        isNewTeam: false,
-      };
-    }
+    if (team) {
+      // If the domain is allowed then we want to assign to the existing team,
+      // otherwise we prevent the creation of another team on self-hosted
+      // instances.
+      if (domain && isDomainAllowed(domain)) {
```

Comment:
isDomainAllowed returns true when no ALLOWED_DOMAINS value is set, this is going to be a change in behavior for existing installations. I don't yet have the best answer for this but it's not quite mergable as it stands

Reply:
Couldn't you add a flag in the .env like MULTI_AUTH=true or false, that defaults to false ?
Existing installation behavior wouldn't change as long as MULTI_AUTH is not set or set to false ?
I understand it's not ideal to add new .env variables but just trying to understand what's blocking that PR

Reply:
I definitely don't want to split authentication codepaths like this, it will lead to security bugs in the long run and more unnecessary config should always be a last resort.

If we get to this (https://github.com/outline/outline/issues/3412) within the next month then it might be a nice middleground – and at this point we could easily advise that all self-hosted installations add in allowed domains in team settings after upgrading.

Reply:
Many thanks for your answer, and very nice to see that you already have a solution in mind for the "multi-auth" case !